src/wh_client_crypto.c

```diff
@@ -2072,4 +2072,492 @@ int wh_Client_CmacCancelableResponse(whClientContext* c, Cmac* cmac,
     return ret;
 }
 #endif /* WOLFSSL_CMAC */
+
+#ifdef HAVE_DILITHIUM
+
+int wh_Client_MlDsaSetKeyId(MlDsaKey* key, whKeyId keyId)
+{
+    if (key == NULL) {
+        return WH_ERROR_BADARGS;
+    }
+
+    key->devCtx = (void*)((intptr_t)keyId);
+    return WH_ERROR_OK;
+}
+
+int wh_Client_MlDsaGetKeyId(MlDsaKey* key, whKeyId* outId)
+{
+    if (key == NULL || outId == NULL) {
+        return WH_ERROR_BADARGS;
+    }
+
+    *outId = (whKeyId)((intptr_t)key->devCtx);
+    return WH_ERROR_OK;
+}
+
+int wh_Client_MlDsaImportKey(whClientContext* ctx, MlDsaKey* key,
+                             whKeyId* inout_keyId, whNvmFlags flags,
+                             uint16_t label_len, uint8_t* label)
+{
+    int      ret    = WH_ERROR_OK;
+    whKeyId  key_id = WH_KEYID_ERASED;
+    byte     buffer[DILITHIUM_MAX_PRV_KEY_SIZE];
+    uint16_t buffer_len = 0;
+
+    if ((ctx == NULL) || (key == NULL) ||
+        ((label_len != 0) && (label == NULL))) {
+        return WH_ERROR_BADARGS;
+    }
+
+    if (inout_keyId != NULL) {
+        key_id = *inout_keyId;
+    }
+
+    ret = wh_Crypto_MlDsaSerializeKeyDer(key, sizeof(buffer), buffer,
+                                         &buffer_len);
+#ifdef DEBUG_CRYPTOCB_VERBOSE
+    printf("[client] %s serialize ret:%d, key:%p, max_size:%u, buffer:%p, "
+           "outlen:%u\n",
+           __func__, ret, key, (unsigned int)sizeof(buffer), buffer,
+           buffer_len);
+#endif
+    if (ret == WH_ERROR_OK) {
+        /* Cache the key and get the keyID */
+        ret = wh_Client_KeyCache(ctx, flags, label, label_len, buffer,
+                                 buffer_len, &key_id);
+        if ((ret == WH_ERROR_OK) && (inout_keyId != NULL)) {
+            *inout_keyId = key_id;
+        }
+    }
+
+#ifdef DEBUG_CRYPTOCB_VERBOSE
+    printf("[client] %s label:%.*s ret:%d keyid:%u\n", __func__, label_len,
+           label, ret, key_id);
+#endif
+    return ret;
+}
+
+int wh_Client_MlDsaExportKey(whClientContext* ctx, whKeyId keyId, MlDsaKey* key,
+                             uint16_t label_len, uint8_t* label)
+{
+    int ret = WH_ERROR_OK;
+    /* buffer cannot be larger than MTU */
+    byte     buffer[DILITHIUM_MAX_PRV_KEY_SIZE];
+    uint16_t buffer_len = sizeof(buffer);
+
+    if ((ctx == NULL) || WH_KEYID_ISERASED(keyId) || (key == NULL)) {
+        return WH_ERROR_BADARGS;
+    }
+
+    /* Now export the DER key from the server */
+    ret =
+        wh_Client_KeyExport(ctx, keyId, label, label_len, buffer, &buffer_len);
+    if (ret == WH_ERROR_OK) {
+        /* Update the key structure */
+        ret = wh_Crypto_MlDsaDeserializeKeyDer(buffer, buffer_len, key);
+    }
+
+#ifdef DEBUG_CRYPTOCB_VERBOSE
+    printf("[client] %s keyid:%x key:%p ret:%d label:%.*s\n", __func__, keyId,
+           key, ret, (int)label_len, label);
+#endif
+    return ret;
+}
+
+static int _MlDsaMakeKey(whClientContext* ctx, int size, int level,
```

Comment:
Can you add a declaration of this at the top?  I swear we were getting complaints about this in the HighTec compiler

Reply:
fixed in #86